excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/validation/CategoryValidator.java

```diff
@@ -0,0 +1,161 @@
+package uk.gov.hmcts.ccd.definition.store.excel.validation;
+
+import com.microsoft.applicationinsights.boot.dependencies.apachecommons.lang3.StringUtils;
+import com.microsoft.applicationinsights.boot.dependencies.apachecommons.lang3.tuple.Pair;
+import com.microsoft.applicationinsights.boot.dependencies.apachecommons.lang3.tuple.Triple;
+import lombok.val;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.definition.store.excel.endpoint.exception.InvalidImportException;
+import uk.gov.hmcts.ccd.definition.store.excel.parser.ParseContext;
+import uk.gov.hmcts.ccd.definition.store.excel.parser.model.DefinitionDataItem;
+import uk.gov.hmcts.ccd.definition.store.excel.util.mapper.ColumnName;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.CaseTypeEntity;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.groupingBy;
+
+@Component
+public class CategoryValidator {
+
+    private ParseContext parseContext;
+    private static final String ERROR_MESSAGE = "CategoryTab Invalid ";
+    private static final String NOT_VALID = " is not a valid ";
+    private static final String DISPLAY_ORDER_ERROR = "DisplayOrder cannot be duplicated within case type"
+        + " category and parent category in Category tab";
+
+
+    public CategoryValidator() {
+    }
+
+    public void validate(ParseContext parseContext, List<DefinitionDataItem> categoryItems) {
+        this.parseContext = parseContext;
+        validateUniqueCategoriesIds(categoryItems);
+        validateUniqueDisplayOrder(categoryItems);
+        val pairCaseTypeCategory = createCategoryIdCaseTypePair(categoryItems);
+        categoryItems.forEach(element -> validate(element, pairCaseTypeCategory));
+    }
+
+    private void validate(DefinitionDataItem definitionDataItem,Map<String,String> pairCaseTypeCategory) {
+        validateCaseType(definitionDataItem);
+        validateCategoryId(definitionDataItem);
+        validateCategoryLabel(definitionDataItem);
```

Comment:
For just checking that mandatory columns are populated, I think we can possibly delete any checks in here and instead add all the mandatory ones into [ColumnName.isRequired()](https://github.com/hmcts/ccd-definition-store-api/blob/85a01eb738281e3f62d2ac6c3f1d8d5b79a5424b/excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/util/mapper/ColumnName.java#L133)?